packages/protocol/test/common/metatransactionwallet.ts

```diff
@@ -300,24 +339,80 @@ contract('MetaTransactionWallet', (accounts: string[]) => {
 
       describe('when the caller is the signer', () => {
         describe('when the transactions are executed directly', () => {
-          beforeEach(async () => {
-            await wallet.executeTransactions(
+          describe('', () => {
+            beforeEach(async () => {
+              await wallet.executeTransactions(
+                transactions.map((t) => t.destination),
+                transactions.map((t) => t.value),
+                ensureLeading0x(transactions.map((t) => trimLeading0x(t.data)).join('')),
+                transactions.map((t) => trimLeading0x(t.data).length / 2),
+                { from: signer }
+              )
+            })
+
+            it('should execute the transactions', async () => {
+              assert.equal(await web3.eth.getBalance(transactions[0].destination), value)
+              assert.equal(await web3.eth.getBalance(transactions[2].destination), value)
+              assert.equal(await wallet.signer(), nonSigner)
+            })
+
+            it('should not increment the nonce', async () => {
+              assertEqualBN(await wallet.nonce(), 0)
+            })
+          })
+
+          it('returns the proper values', async () => {
+            const boolTrue: string = '01'
+            const signerAddrReturned: string = trimLeading0x(signer).toLowerCase()
+            const returnValues = await wallet.executeTransactions.call(
               transactions.map((t) => t.destination),
               transactions.map((t) => t.value),
               ensureLeading0x(transactions.map((t) => trimLeading0x(t.data)).join('')),
               transactions.map((t) => trimLeading0x(t.data).length / 2),
               { from: signer }
             )
+
+            assert.equal(
+              returnValues[0],
+              // return values are padded to equal 32 bytes
+              `0x${signerAddrReturned.padStart(64, '0')}${boolTrue.padStart(64, '0')}`
+            )
+            for (let i in returnValues[1]) {
+              // tx's with index 3 and 4 have return values
+              if (parseInt(i) == 3 || parseInt(i) == 4) {
```

Comment:
can we initialize these constants where we initialize the transactions

Reply:
I assigned these numbers to constants within the test which refer to what these numbers actually mean. This adds to the readability. Let me know if this addresses your concerns